### ISSUE TITLE
Add a "Close Right" in menu

### ISSUE THREAD

**Issue by fredericch** [apache/netbeans#4710]
### Description

Add a "Close Right" function on the menu along "Close", "Close All" and "Close Other".

When at least 2 files have been opened, we could close all windows on the right of the current active one.

![image](https://user-images.githubusercontent.com/11049456/193430514-3cc97024-2c50-4000-a2ee-6c99fad7b78b.png)

It would be the same implementation as in Chrome or FF.

### Use case/motivation

When doing a search, I often open "new" windows which are actually not currently usefull and it clutters the tabs. So I spend a lot of time closing them individually. It would save a lot of time to be able to close them all at once.

I cannot use "Close Other" or "Close All", because the tabs on the left are usually the ones I am working on.

When a lot of windows have been opened if the search was too generic, I use this work around:
- Close All
- Open all modified files from the Git menu

### Are you willing to submit a pull request?

No

### Code of Conduct

Yes

**Comment by mbien** [comment]
makes sense. Added [good first issue](https://github.com/apache/netbeans/labels/good%20first%20issue) label in case someone wants to implement it.

I just checked how FF solved this and it has `Close Tabs to Left`, `.. Right` and `Close Other Tabs` in a sub menu.

NetBeans however also supports multi row tabs which would make this a bit more difficult to implement.

**Comment by 6PATyCb** [comment]
When too much files is opened I like to use `documents list` format for manage it:
![image](https://user-images.githubusercontent.com/33724393/194762879-ea2250c6-222e-429d-8f59-c1b70390c588.png)

but in that format would be good to have close button to group of documents:
![image](https://user-images.githubusercontent.com/33724393/194763097-3fa38dee-932c-4e11-a956-de12cda764b7.png)

**Comment by mbien** [comment]
@6PATyCb thats a good idea but next time, please open a new issue. I moved it to 4760

**Comment by 6PATyCb** [comment]
> @6PATyCb thats a good idea but next time, please open a new issue. I moved it to #4760

I thought this was just part of the discussion about usability of documents tabs, but since it looks good as a separate feature, why not :)

**Comment by tc430gi** [comment]
I would like to implement it for you like my good first issue, need it for school credit at University.

**Comment by tc430gi** [comment]
@mbien can you please?

**Comment by 6PATyCb** [comment]
> Hello can @fredericch, @6PATyCb, or anyone assign me this issue please?

I don't have a rights

**Comment by mbien** [comment]
you can simply open a PR if you want to contribute a change. You don't even need an issue for that or someone assigning an issue to you.

**Comment by tc430gi** [comment]
Can somebody help me navigate a little bit in the project or give me some advice where to start?

**Comment by tc430gi** [comment]
Can you help me navigate a little bit in the project? @mbien

**Comment by mbien** [comment]
@tc430gi I don't know that area well but after searching a bit around:

the logic for closing tabs seems to be here:
https://github.com/apache/netbeans/blob/2c27f5a31b64dc0a8e3ddbd08a3223f7ed1d141e/platform/core.windows/src/org/netbeans/core/windows/view/ui/TabbedHandler.java#L317-L333

actions are setup here (and the method below too):
https://github.com/apache/netbeans/blob/c084119009d2e0f736f225d706bc1827af283501/platform/core.windows/src/org/netbeans/core/windows/actions/ActionUtils.java#L60-L77

If you need more info it is probably better to ask on the dev list or open a discussion on github.

**Comment by eirikbakke** [comment]
Once https://github.com/apache/netbeans/pull/4792 merges I'll submit a PR which does a few changes in related areas in ActionUtils.

Most likely you'd need a new action similar to CloseAllButThisAction and CloseAllDocumentsAction.

**Comment by mbien** [comment]
this plugin sounds like it is covering exactly this features:

https://plugins.netbeans.apache.org/catalogue/?id=78
